### PR TITLE
LibJS+LibUnicode: Begin implementing `Intl.DateTimeFormat`

### PR DESCRIPTION
**Pull request by trflynn89**
This implements the constructor and the common Intl methods `supportedLocalesOf` and `resolvedOptions`.
I need a nap :laughing:

### CODE REVIEW

**Comment by trflynn89** [comment]
Reg #9633

**Comment by trflynn89** [comment]
CI failures are legit:
```
UBSAN: reference binding to address 0x01ef2bc4 with insufficient space for type 'const AK::Array<AK::StringView, 0>'
UBSAN: at /home/vsts/work/1/s/Userland/Libraries/LibJS/Runtime/Intl/DateTimeFormat.h, line 182, column: 9
UB is configured to be deadly
ASSERTION FAILED: false
```

File: Userland/Libraries/LibJS/Runtime/Intl/DateTimeFormat.cpp [hunk 1]
```diff
@@ -51,4 +55,616 @@ StringView DateTimeFormat::style_to_string(Style style)
...
+                auto options_prop_index = static_cast<int>(find_index(values.begin(), values.end(), *options_prop));
+
+                // 4. Let formatPropIndex be the index of formatProp within values.
+                auto format_prop_index = static_cast<int>(find_index(values.begin(), values.end(), *format_prop));
+
+                // 5. Let delta be max(min(formatPropIndex - optionsPropIndex, 2), -2).
+                int delta = max(min(format_prop_index - options_prop_index, 2), -2);
+
+                // 6. If delta = 2, decrease score by longMorePenalty.
+                if (delta == 2) {
```
Comment (linusg) [inline Userland/Libraries/LibJS/Runtime/Intl/DateTimeFormat.cpp:629]: could omit the curly braces here

**Review by linusg — COMMENTED** [review]
Typo in ` LibJS: Implement a nearly empty Intl.DateTimeFormat object`:

> Note that unlike other Intl object,

Other than that, this is perfect :tada:
And given that it's almost 3k lines, that's *really* impressive. Glad you decided to work on LibUnicode & Intl, I think we couldn't ask for a better implemenation :^)

**Comment by trflynn89** [comment]
:tada: fixed!

**Review by linusg — APPROVED** [review]

**Pull request merged**